### PR TITLE
Remove disabled advisers from event organiser field.

### PR DESCRIPTION
**Pull request by ztolley**
Made some improvement to the adviser repo and tests

Refactored the event sub app to get all advisers for filter forms and filter out disabled advisers in the organiser field in the edit/create form.

### CODE REVIEW

File: src/apps/adviser/repos.js [hunk 1]
```diff
@@ -9,9 +9,14 @@ const authorisedRequest = require('../../lib/authorised-request')
...
 function getAdvisers (token) {
   return authorisedRequest(token, `${config.apiRoot}/adviser/?limit=100000&offset=0`)
-    .then(data => assign({}, data, {
-      results: data.results.filter(adviser => Boolean(adviser.first_name || adviser.last_name)),
-    }))
+    .then(response => {
+      const results = response.results.filter(adviser => get(adviser, 'name', '').trim().length)
+
+      return {
+        count: results.length,
```
Comment (teneightfive) [inline src/apps/adviser/repos.js]: 👍

File: src/apps/adviser/repos.js [hunk 2]
```diff
@@ -9,9 +9,14 @@ const authorisedRequest = require('../../lib/authorised-request')
...
+        results,
```
Comment (teneightfive) [inline src/apps/adviser/repos.js]: Swap the order of these. We use the pattern of shorthand declared first for easier reading.

File: src/apps/events/controllers/edit.js [hunk 3]
```diff
@@ -3,30 +3,42 @@ const { eventFormConfig } = require('../macros')
...
-  const lead_team = eventData.lead_team || get(req, 'session.user.dit_team.id')
-  const mergedData = pickBy(merge({}, eventData, { lead_team }, res.locals.requestBody))
-  const eventForm =
-    buildFormWithStateAndErrors(
-      eventFormConfig({ eventId, advisers }),
-      mergedData,
-      get(res.locals, 'form.errors.messages'),
-    )
+async function getActiveAdvisers (token, currentAdviser) {
+  const allAdvisers = await getAdvisers(token)
```
Comment (teneightfive) [inline src/apps/events/controllers/edit.js]: try/catch around this?

File: src/apps/adviser/repos.js (hunk 2 above)
Comment (ztolley) [reply src/apps/adviser/repos.js]: Wish there was a lint rule for that

File: src/apps/events/controllers/edit.js (hunk 3 above)
Comment (ztolley) [reply src/apps/events/controllers/edit.js]: It gets caught in the renderEditPage function, I tested it.

File: src/apps/adviser/repos.js (hunk 2 above)
Comment (teneightfive) [reply src/apps/adviser/repos.js]: Yeh, I looked before but couldn't find one for eslint. Think there is one for JSHint though.

File: src/apps/events/controllers/edit.js (hunk 3 above)
Comment (teneightfive) [reply src/apps/events/controllers/edit.js]: Ah I see because its not being called as part of middleware like we've used elsewhere.

Comment (ztolley) [reply src/apps/events/controllers/edit.js]: Trying not to overdo middleware, in this case it's just split out to a function to try and keep the code clean and follow the mantra 'Do one thing and do it well'

**Pull request merged**